File: pdlearning/src/microservices/uploader-microservice/Api/Application/Models/PersonalFileModel.cs
using System;
using Microservice.Uploader.Domain.Entities;
using Microservice.Uploader.Domain.ValueObjects;

namespace Microservice.Uploader.Application.Models
{
    public class PersonalFileModel
    {
        public PersonalFileModel()
        {
        }

        public PersonalFileModel(PersonalFile entity)
        {
            this.Id = entity.Id;
            this.UserId = entity.UserId;
            this.FileName = entity.FileName;
            this.FileType = entity.FileType;
            this.FileExtension = entity.FileExtension;
            this.FileSize = entity.FileSize;
            this.FileLocation = entity.FileLocation;
            this.CreatedDate = entity.CreatedDate;
        }

        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public string FileName { get; set; }

        public FileType FileType { get; set; }

        public string FileExtension { get; set; }

        public double FileSize { get; set; }

        public string FileLocation { get; set; }

        public DateTime CreatedDate { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/personal-space/models/personal-file.model.ts
import { FileType } from './file-type.enum';

export enum PersonalFileQueryMode {
  All = 'All'
}

export const PERSONAL_FILE_QUERY_MODE_LABEL = new Map<PersonalFileQueryMode, string>([[PersonalFileQueryMode.All, 'Files']]);

export interface IPersonalFileModel {
  id: string;
  fileName: string;
  fileType: FileType;
  fileSize: number;
  fileLocation: string;
  fileExtension: string;
  createdDate?: Date;
}
export class PersonalFileModel implements IPersonalFileModel {
  public id: string;
  public fileName: string;
  public fileType: FileType;
  public fileSize: number;
  public fileLocation: string;
  public fileExtension: string;
  public createdDate?: Date;

  constructor(data?: IPersonalFileModel) {
    if (data != null) {
      this.id = data.id;
      this.fileName = data.fileName;
      this.fileType = data.fileType;
      this.fileSize = data.fileSize;
      this.fileLocation = data.fileLocation;
      this.fileExtension = data.fileExtension;
      this.createdDate = data.createdDate;
    }
  }
}
